### DISCUSSION TITLE
Table definition, reference to non (int) id column

### DISCUSSION THREAD

**Discussion by blastiko** [wheels-dev/wheels#1149] [Q&A]
Hi all
How can i make a reference column to a table with no auto-increment (column) id but with a string column for primary key?

This is the table with the string primary key (abbreviation):
`t = createTable(name = 'districts', id=false);
t.primaryKey(name = 'abbreviation', type='string', null=false, limit='2');
t.string(columnNames='distrName', null=false, limit='128');
t.integer(columnNames="codRegion", null=false, limit='2');
t.timestamps();
t.create();
addIndex(table = "districts", columnNames = "abbreviation", unique = true);`

In another table, how can i set a reference column to 'abbreviation' on 'district' table?

Thanks

**Comment by bpamiri** [comment]
What does your model look like for the district table?

Look at the section [Breaking the Convention](https://guides.cfwheels.org/cfwheels-guides/database-interaction-through-models/associations#breaking-the-convention) which may give you the answer.

**Comment by blastiko** [comment]
Hi Peter,
I think i have solved it with this code:
`addForeignKey(table = "cities", referenceTable = "districts", column = "abbreviation", referenceColumn = "abbreviation", onUpdate="cascade", onDelete="restrict");`

But, is it the right way?

**Reply by bpamiri** [reply]
Yes I think that is fine. The addReference() method:
```
	public void function addReference(required string table, required string referenceName) {
		addForeignKey(
			table = arguments.table,
			referenceTable = pluralize(arguments.referenceName),
			column = "#arguments.referenceName#id",
			referenceColumn = "id"
		);
```

just makes a call to the addForeignKey() method and defaults the referenceColumn to "id". So this way you are able to define your own referenceColumn name.
